ziti-springboot-client/src/main/java/org/openziti/springboot/client/web/config/ZitiHttpClientConfiguration.java

```diff
@@ -42,126 +40,125 @@
 public class ZitiHttpClientConfiguration {
 
   // The default timeout when requesting a connection from the connection manager.
-  private static final long DEFAULT_CONNECTION_REQUEST_TIMEOUT = 30000;
+  private static final long DEFAULT_CONNECTION_REQUEST_TIMEOUT = 30 * 1000;
 
   // The default time to wait for data, the maximum time between two consecutive packets of data.
-  private static final int DEFAULT_RESPONSE_TIMEOUT = 60000;
+  private static final int DEFAULT_RESPONSE_TIMEOUT = 60 * 1000;
 
   // The default time to keep a connection alive.
   private static final long DEFAULT_KEEP_ALIVE_TIME_MILLIS = 20 * 1000;
 
   private ZitiConnectionSocketFactory zitiConnectionSocketFactory;
   private ZitiSSLConnectionSocketFactory zitiSSLConnectionSocketFactory;
 
-  @ConditionalOnMissingBean(name = "zitiRestTemplate")
-  @Bean("zitiRestTemplate")
-  public RestTemplate restTemplate(@Qualifier("zitiRestTemplateBuilder") RestTemplateBuilder restTemplateBuilder) {
+  @ConditionalOnProperty(value = "spring.ziti.client.rest-template.enabled", havingValue = "true", matchIfMissing = true)
+  @Bean
+  public RestTemplate zitiRestTemplate(@Qualifier("zitiRestTemplateBuilder") RestTemplateBuilder restTemplateBuilder) {
     return restTemplateBuilder.build();
   }
 
-  @ConditionalOnMissingBean(name = "zitiRestTemplateBuilder")
-  @Bean("zitiRestTemplateBuilder")
-  public RestTemplateBuilder restTemplateBuilder(@Qualifier("zitiHttpClient") HttpClient httpClient) {
+  @ConditionalOnProperty(value = "spring.ziti.client.rest-template-builder.enabled", havingValue = "true", matchIfMissing = true)
+  @Bean
+  public RestTemplateBuilder zitiRestTemplateBuilder(@Qualifier("zitiHttpClient") HttpClient httpClient) {
     return new RestTemplateBuilder()
-        .requestFactory(() -> clientHttpRequestFactory(httpClient));
+        .requestFactory(() -> zitiClientHttpRequestFactory(httpClient));
   }
 
-  @ConditionalOnMissingBean(name = "zitiClientHttpRequestFactory")
-  @Bean("zitiClientHttpRequestFactory")
-  public ClientHttpRequestFactory clientHttpRequestFactory(@Qualifier("zitiHttpClient") HttpClient httpClient) {
-    HttpComponentsClientHttpRequestFactory clientHttpRequestFactory = new HttpComponentsClientHttpRequestFactory();
+  @ConditionalOnProperty(value = "spring.ziti.client.request-factory.enabled", havingValue = "true", matchIfMissing = true)
+  @Bean
+  public ClientHttpRequestFactory zitiClientHttpRequestFactory(@Qualifier("zitiHttpClient") HttpClient httpClient) {
+    final HttpComponentsClientHttpRequestFactory clientHttpRequestFactory = new HttpComponentsClientHttpRequestFactory();
     clientHttpRequestFactory.setHttpClient(httpClient);
     return clientHttpRequestFactory;
   }
 
-  @ConditionalOnMissingBean(ZitiConnectionSocketFactory.class)
+  @ConditionalOnProperty(value = "spring.ziti.client.context.enabled", havingValue = "true", matchIfMissing = true)
   @Bean
-  public ZitiConnectionSocketFactory zitiConnectionSocketFactory(
```

Comment:
Are you removing this function too? Is this intentional?

Reply:
I don't think this is deleted, just moved further down in the file. See line 111-122.